kostal/integrationpluginkostal.cpp

```diff
@@ -0,0 +1,404 @@
+            });
+
+            connect(kostalConnection, &KostalConnection::activePowerPhase3Changed, this, [thing](float activePowerPhase3){
+                qCDebug(dcKostal()) << thing << "active power phase 3 changed" << activePowerPhase3 << "W";
+                thing->setStateValue(kostalInverterCurrentPowerPhaseCStateTypeId, activePowerPhase3);
+            });
+
+
+            // Update registers
+            kostalConnection->update();
+        });
+
+        connect(kostalConnection, &KostalConnection::connectionStateChanged, this, [this, thing, kostalConnection](bool status){
+            qCDebug(dcKostal()) << "Connected changed to" << status << "for" << thing;
+            if (status) {
+                // Connected true will be set after successfull init
+                kostalConnection->initialize();
+            } else {
+                thing->setStateValue(kostalInverterConnectedStateTypeId, false);
+                foreach (Thing *childThing, myThings().filterByParentId(thing->id())) {
+                    if (childThing->thingClassId() == kostalBatteryThingClassId) {
+                        childThing->setSettingValue(kostalBatteryConnectedStateTypeId, false);
+                    } else if (childThing->thingClassId() == kostalMeterThingClassId) {
+                        childThing->setSettingValue(kostalMeterConnectedStateTypeId, false);
+                    }
+                }
+            }
+        });
+    }
+
+    if (thing->thingClassId() == kostalMeterThingClassId) {
+        // Get the parent thing and the associated connection
+        Thing *connectionThing = myThings().findById(thing->parentId());
+        if (!connectionThing) {
+            qCWarning(dcKostal()) << "Failed to set up kostal energy meter because the parent thing with ID" << thing->parentId().toString() << "could not be found.";
+            info->finish(Thing::ThingErrorHardwareNotAvailable);
+            return;
+        }
+
+        KostalConnection *kostalConnection = m_kostalConnections.value(connectionThing);
+        if (!kostalConnection) {
+            qCWarning(dcKostal()) << "Failed to set up kostal energy meter because the connection for" << connectionThing << "does not exist.";
+            info->finish(Thing::ThingErrorHardwareNotAvailable);
+            return;
+        }
+
+        // Note: The connected state will be handled in the parent inverter thing
+
+        // Update the meter data from the kostal connection containing all information
+
+        connect(kostalConnection, &KostalConnection::powerMeterTotalActivePowerChanged, this, [thing](float powerMeterTotalActivePower){
+            qCDebug(dcKostal()) << thing << "total active power changed" << powerMeterTotalActivePower << "W";
+            thing->setStateValue(kostalMeterCurrentPowerStateTypeId, powerMeterTotalActivePower);
+        });
+
+        // TODO: set toal energy consumed/produced
```

Comment:
too much for now?

Reply:
untagged the entire branch, moved the requied stuff to land into #44 and rebased all sub branches. Will continue after landing and once I have access to a test device.